pkg/rpcinfo/remoteinfo/remoteInfo.go

```diff
@@ -48,7 +48,7 @@ type RemoteInfo interface {
 
 // RemoteAddrSetter is used to set remote addr.
 type RemoteAddrSetter interface {
-	SetRemoteAddr(addr net.Addr) (ok bool)
+	SetRemoteAddr(addr net.Addr) discovery.Instance
```

Comment:
add comment for this method, why return discovery.Instance. it is better to rename, eg
`GetNewInstanceWithAddr() `